src/Planners/include/BipedalLocomotion/Planners/SO3Planner.tpp

```diff
@@ -0,0 +1,107 @@
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+
+template <Representation representation>
+bool SO3Planner<representation>::setRotations(const manif::SO3d& initialRotation,
+                                              const manif::SO3d& finalRotation,
+                                              const double& T)
+{
+    if (T <= 0)
+    {
+        std::cerr << "[SO3Planner::setRotation] The trajectory duration must be a strictly "
+                     "positive number."
+                  << std::endl;
+        return false;
+    }
+
+    m_initialRotation = initialRotation;
+    m_T = T;
+
+    if constexpr (representation == Representation::RightTrivialized)
+    {
+        m_distance = (finalRotation * initialRotation.inverse()).log();
+    } else
+    // Please read it as representation == Representation::LeftTrivialized
+    {
+        m_distance = (initialRotation.inverse() * finalRotation).log();
+    }
+
+    return true;
+}
+
+template <Representation representation>
+template <class Derived>
+bool SO3Planner<representation>::evaluatePoint(const double& time,
+                                               manif::SO3d& rotation,
+                                               manif::SO3TangentBase<Derived>& velocity,
+                                               manif::SO3TangentBase<Derived>& acceleration) const
+{
+    if (time < 0 || time > m_T)
+    {
+        std ::cerr << "[SO3Planner::evaluatePoint] The time has to be a real positive number "
+                      "between 0 and "
+                   << m_T << "." << std::endl;
+        return false;
+    }
+
+    const double& t = time;
+    // Compute the displacement in the tangent space
+    // - displacement_tangent = log(finalRotation * initialRotation.inverse()) * s(t) (in case of
+    //                                                                                 Right
+    //                                                                                 Trivialization)
+    // - displacement_tangent = log(initialRotation.inverse() * finalRotation) * s(t) (in case of
+    //                                                                                 Left
+    //                                                                                 Trivialization)
+    const double s = 10 * std::pow(t / m_T, 3)
+                     - 15 * std::pow(t / m_T, 4)
+                     + 6 * std::pow(t / m_T, 5);
+    const manif::SO3d::Tangent displacementTangent = m_distance * s;
+
+    if constexpr (representation == Representation::RightTrivialized)
+    {
+        // please read this as R(t) = exp(displacementTangent) * initialRotation
+        rotation = displacementTangent + m_initialRotation;
+    } else
+    // Please read it as representation == Representation::LeftTrivialized
+    {
+        // please read this as R(t) = initialRotation * exp(displacementTangent)
+        rotation = m_initialRotation + displacementTangent;
```

Comment:
Even if it is the same, is it possible to use the version you put in the comments? It hurts me that the commutative property does not hold with the ``+`` operator. :sweat_smile:

Reply:
Done here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/84/commits/6875b42e57335a0a0fbcaf2fb620a89cc20172e7